### PR TITLE
CMR-10139: Updated code to remove subscription information from redis cache upon subscription delete. Updated schema to allow multiple modes.

### PR DESCRIPTION
**Pull request by eereiter**
# Overview

### What is the feature/fix?

Updated code to remove subscription information from redis cache upon subscription delete. Updated schema to allow multiple modes. Adding subscription to redis cache already in the code. 

### What is the Solution?

Updated code to remove subscription information from redis cache upon subscription delete. Updated schema to allow multiple modes.

### What areas of the application does this impact?

Metadata_db

# Checklist

- [X] I have updated/added unit and int tests that prove my fix is effective or that my feature works - in previous commit
- [X] New and existing unit and int tests pass locally and remotely
- [X] clj-kondo has been run locally and all errors corrected
- [ ] I have removed unnecessary/dead code and imports in files I have changed
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have cleaned up integration tests by doing one or more of the following:
  - migrated any are2 tests to are3 in files I have changed
  - de-duped, consolidated, removed dead int tests
  - transformed applicable int tests into unit tests
  - refactored to reduce number of system state resets by updating fixtures (use-fixtures :each (ingest/reset-fixture {})) to be :once instead of :each

### CODE REVIEW

File: metadata-db-app/src/cmr/metadata_db/services/concept_service.clj [hunk 1]
```diff
@@ -776,15 +777,29 @@
...
+                               (= :subscription concept-type))
                          (:metadata previous-revision)
                          "")
-              tombstone (merge previous-revision {:concept-id concept-id
-                                                  :revision-id revision-id
-                                                  :revision-date revision-date
-                                                  :user-id user-id
-                                                  :metadata metadata
-                                                  :deleted true})]
+              tombstone (if (= :subscription concept-type)
```
Comment (TylerHeald1) [inline metadata-db-app/src/cmr/metadata_db/services/concept_service.clj:784]: Does this update not need to be done for the other save-concept-revision? Is it only relevant for tombstones?

File: metadata-db-app/src/cmr/metadata_db/services/subscription_cache.clj [hunk 2]
```diff
@@ -4,8 +4,11 @@
 	<collection-concept-id> --> <ingest subscription map>
 
 	Example:
-	'C12344554-PROV1' --> { 'enabled' true|false
-                          'mode'    New|Update|Delete|All} "
+  {Collection concept id 1: {\"New\" 1
```
Comment (TylerHeald1) [inline metadata-db-app/src/cmr/metadata_db/services/subscription_cache.clj:7]: Does using different numbers here (after new/update/delete) affect anything?

Comment (TylerHeald1) [reply metadata-db-app/src/cmr/metadata_db/services/subscription_cache.clj:7]: Oh I see, it's the count of how many of those type of subscriptions there are

**Review by TylerHeald1 — APPROVED** [review]

File: metadata-db-app/src/cmr/metadata_db/services/concept_service.clj (hunk 1 above)
Comment (jmaeng72) [inline metadata-db-app/src/cmr/metadata_db/services/concept_service.clj:784]: I think this will be more readable as a helper func

Comment (eereiter) [reply metadata-db-app/src/cmr/metadata_db/services/concept_service.clj:784]: Fixed.

**Pull request merged**